lib/services/unified_resource.go

```diff
@@ -233,43 +215,89 @@ type iteratedItem struct {
 // method.
 func (c *UnifiedResourceCache) iterateItems(ctx context.Context, start string, sortBy types.SortBy, kinds ...string) iter.Seq2[iteratedItem, error] {
 	return func(yield func(iteratedItem, error) bool) {
-		startKey := getStartKey(start, sortBy)
-
 		kindsMap := make(map[string]struct{})
 		for _, k := range kinds {
 			kindsMap[k] = struct{}{}
 		}
 
-		err := c.read(ctx, func(cache *UnifiedResourceCache) error {
-			tree, err := cache.getSortTree(sortBy.Field)
-			if err != nil {
-				return trace.Wrap(err, "getting sort tree")
-			}
+		var startKey backend.Key
+		switch {
+		case start != "":
+			startKey = backend.KeyFromString(start)
+		case start == "" && sortBy.IsDesc:
+			startKey = backend.RangeEnd(backend.NewKey(prefix))
+		default:
+			startKey = backend.NewKey(prefix)
+		}
 
-			iterateRange := tree.DescendRange
-			endKey := backend.NewKey(prefix)
-			if !sortBy.IsDesc {
-				iterateRange = tree.AscendRange
-				endKey = backend.RangeEnd(endKey)
-			}
+		endKey := backend.RangeEnd(backend.NewKey(prefix))
+		compareDirection := 1
+		itemIter := (*btree.BTreeG[*item]).AscendGreaterOrEqual
+		if sortBy.IsDesc {
+			endKey = backend.NewKey(prefix)
+			compareDirection = -1
+			itemIter = (*btree.BTreeG[*item]).DescendLessOrEqual
+		}
```

Comment:
This is very confusing to follow and I'm not sure why we're using a key prefix but we should check that the caller-provided `start` isn't before the beginning of the full iteration range.